### ISSUE TITLE
-Wbitfield-constant-conversion does not warn on 1-bit signed integer bitfield edge case

### ISSUE THREAD

**Issue by Scaless** [llvm/llvm-project#53253]
In C++, signed integers are represented in two's complement. This also applies to signed bitfields.
A signed bitfield composed of one bit can therefore store a value in the range -1 to 0. 
Assigning a value of 1 to such a bitfield should produce a warning since it is out of range of representable values.

MSVC and GCC generate the correct warnings for this situation, clang does not.

```
clang 13.0.0
-std=c++20 -Wall -Wextra -Wpedantic -Wconversion -Wbitfield-constant-conversion

struct test
{
    signed int v : 1;
};

int main()
{
    test t;
    // MSVC:  warning C4463: overflow; assigning 1 to bit-field that can only hold values from -1 to 0
    // GCC:   overflow in conversion from 'int' to 'signed char:1' changes value from '1' to '-1' [-Woverflow]
    // clang: no warning produced
    t.v = 1;
    return t.v; // Returns -1
}
```

[Godbolt](https://godbolt.org/z/dM7vsxPYb)

**Comment by Scaless** [comment]
Looks like the root cause is this section here: https://github.com/llvm/llvm-project/blob/ba6485e25fc56468f34cc8a6938d66d3c5f46596/clang/lib/Sema/SemaChecking.cpp#L12468-L12471

Removing those lines gives me the proper warning: 

![image](https://user-images.githubusercontent.com/3276824/149850596-a55dcd3b-e4df-463e-99e1-08049a36e8cf.png)

Boolean types are already excluded from analysis earlier in the function so removing would not affect cases such as `bool b : 1;`:
https://github.com/llvm/llvm-project/blob/ba6485e25fc56468f34cc8a6938d66d3c5f46596/clang/lib/Sema/SemaChecking.cpp#L12358-L12361

The original code's assumption seems to be that `signed int x : 1 = true;` should not give a warning, which I (and GCC/MSVC) disagree with. If you want a boolean true/false (1/0) you should be using a `bool : 1` or `unsigned int : 1` which stores the proper value.

**Comment by AaronBallman** [comment]
Note, this was further updated by 3b00e486797c0a28f0b5216ea507329d098ce573.
